tests/unit_tests/weightwindows/test_ww_gen.py

```diff
@@ -0,0 +1,271 @@
+        analog_mean = np.copy(lib_tally.mean)
+
+        # update the weight window values using tally results
+        ww.update_magic(lib_tally)
+
+        print(ww.bounds[0].max())
+        print(ww.bounds[0].min())
+        assert any(ww.bounds[0] != -1)
+        assert any(ww.bounds[1] != -1)
+
+        # make sure that the weight window update doesn't change tally values
+        np.testing.assert_equal(lib_tally.mean, analog_mean)
+
+        # check against weight windows from the previous iteration
+        # the order of filters should not change the weight window values
+        if ref_lower is None:
+            ref_lower = np.copy(ww.bounds[0])
+        else:
+            np.testing.assert_equal(ref_lower, ww.bounds[0])
+
+        if ref_upper is None:
+            ref_upper = np.copy(ww.bounds[1])
+        else:
+            np.testing.assert_equal(ref_upper, ww.bounds[1])
+
+        # turn on weight windows for the subsequent run
+        openmc.lib.settings.weight_windows_on = True
+
+        openmc.lib.hard_reset()
+
+        openmc.lib.run()
+
+        ww_mean = np.copy(lib_tally.mean)
+
+        # we expect that the application of weight windows will populate more tally
+        # bins than the analog run for the meshes in this test model
+        assert any(ww_mean != analog_mean)
+        assert np.count_nonzero(ww_mean) > np.count_nonzero(analog_mean)
+
+
+def test_ww_import_export(run_in_tmpdir, model):
+    # create a tally for weight windows
+    mesh = openmc.RegularMesh()
+    mesh.lower_left = [-50.0] * 3
+    mesh.upper_right = [50.0] * 3
+    # use un-equal mesh widths in each dimension to more robustly check
+    # use of tally data
+    mesh.dimension = (3, 4, 5)
+
+    mf = openmc.MeshFilter(mesh)
+
+    e_groups = np.logspace(0, 7, 8)
+    ef = openmc.EnergyFilter(e_groups)
+
+    pf = openmc.ParticleFilter(['neutron', 'photon'])
+
+    tally = openmc.Tally()
+    tally.filters = [mf, ef, pf]
+    tally.scores = ['flux']
+
+    model.tallies = openmc.Tallies([tally])
+
+    # first, generate some weight windows
+    model.export_to_model_xml()
+
+    openmc.lib.init()
+
+    tally = openmc.lib.tallies[1]
+
+    ww = openmc.lib.WeightWindows.from_tally(tally)
+
+    openmc.lib.run()
+
+    mean_before = np.array(tally.mean)
+
+    ww.update_magic(tally)
+
+    mean_after = np.array(tally.mean)
+
+    assert (mean_before == mean_after).all()
+
+    lb_before, up_before = ww.bounds
+
+    openmc.lib.export_weight_windows()
+
+    assert Path('weight_windows.h5').exists()
+
+    openmc.lib.import_weight_windows('weight_windows.h5')
+
+    ww = openmc.lib.weight_windows[2]
+
+    lb_after, up_after = ww.bounds
+
+    assert np.allclose(lb_before, lb_after)
+    assert np.allclose(up_before, up_after)
+
+    openmc.lib.finalize()
+
+
+def test_ww_gen_roundtrip(run_in_tmpdir, model):
+
+    mesh = openmc.RegularMesh.from_domain(model.geometry.root_universe)
+    energy_bounds = np.linspace(0.0, 1E8, 11)
```

Comment:
```suggestion
    energy_bounds = np.linspace(0.0, 1e8, 11)
```

Reply:
Why is the lowercase version preferred?

Reply:
Mostly just prevailing style. But I'd note that lowercase is what Python uses by default, e.g.
```Python
>>> from scipy import constants
>>> constants.hbar
1.0545718176461565e-34
```